test/sagemaker_tests/huggingface_pytorch/training/integration/sagemaker/test_trcomp.py

```diff
@@ -107,106 +136,280 @@ def test_trcomp_default(self, patched, ecr_image, sagemaker_session, tmpdir, py_
+            "branch": "v" + transformers_version,
+        }
+
+        source_dir = (
+            "./examples/question-answering"
+            if Version(transformers_version) < Version("4.6")
+            else "./examples/pytorch/question-answering"
+        )
+
+        hyperparameters["max_steps"] = 3 * num_gpus_per_instance
+
+        with timeout(minutes=DEFAULT_TIMEOUT):
+            estimator = HuggingFace(
+                compiler_config=TrainingCompilerConfig(),
+                entry_point="run_qa.py",
+                source_dir=source_dir,
+                git_config=git_config,
+                metric_definitions=metric_definitions,
+                role="SageMakerRole",
+                image_uri=ecr_image,
+                instance_count=instance_count,
+                instance_type=instance_type,
+                sagemaker_session=sagemaker_session,
+                hyperparameters=hyperparameters,
+                py_version=py_version,
+                max_retry_attempts=15,
+            )
+            estimator.fit(
+                job_name=sagemaker.utils.unique_name_from_base("hf-pt-trcomp-SNMG-default"),
+                logs=True,
+            )
+        captured = capsys.readouterr()
+        logs = captured.out + captured.err
+        assert "Found configuration for Training Compiler" in logs
+        assert "Configuring SM Training Compiler" in logs
+        assert "device: xla" in logs
+
+
+@pytest.mark.integration("sagmaker-training-compiler")
+@pytest.mark.processor("gpu")
+@pytest.mark.skip_py2_containers
+@pytest.mark.skip_huggingface_containers
+@pytest.mark.skip_cpu
+@mock.patch("sagemaker.huggingface.TrainingCompilerConfig.validate", return_value=None)
+class TestMultiNodeMultiGPU:
+    """
+    All Multi Node Multi GPU tests go here.
+    """
+
+    @pytest.mark.parametrize(
+        "instance_type, instance_count",
+        [
+            ("ml.p3.16xlarge", 2),
+            ("ml.p4d.24xlarge", 2),
+        ],
+    )
+    @pytest.mark.model("bert-large")
+    def test_trcomp_default(
+        self,
+        patched,
+        ecr_image,
+        sagemaker_session,
+        tmpdir,
+        py_version,
+        capsys,
+        instance_type,
+        instance_count,
+        num_gpus_per_instance,
+    ):
+        """
+        Tests the default configuration of SM trcomp
+        """
+        transformers_version = get_transformers_version(ecr_image)
+        git_config = {
+            "repo": "https://github.com/huggingface/transformers.git",
+            "branch": "v" + transformers_version,
+        }
+
+        source_dir = (
+            "./examples/question-answering"
+            if Version(transformers_version) < Version("4.6")
+            else "./examples/pytorch/question-answering"
+        )
+
+        hyperparameters["max_steps"] = 3 * num_gpus_per_instance * instance_count
+        hyperparameters["sagemaker_pytorch_xla_multi_worker_enabled"] = True
```

Comment:
nitpick: please add a comment here to explain that this is the same as setting distribution to xla in the estimator.